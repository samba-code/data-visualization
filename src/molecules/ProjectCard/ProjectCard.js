import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { Link } from "@reach/router";
import { lighten, rem } from "polished";
const ArticalCard = styled.article`
  display: flex;
  box-sizing: border-box;
  min-height: 160px;
  flex-grow: 1;
  flex-shrink: 1;
  align-content: flex-start;
  background: ${(props) => lighten("0.2", props.theme.neutral)};
  margin: 26px 13px 0 13px;
  @media only screen and (max-width: 700px) {
    flex-basis: 100%;
  }
  @media only screen and (min-width: 700px) {
    flex-basis: 40%;
  }
  @media only screen and (min-width: 1200px) {
    flex-basis: 30%;
  }
`;

ArticalCard.defaultProps = {
  theme: {
    neutral: "grey",
  },
};

const CardHeading1 = styled.h1`
  font-size: ${rem("18px")};
  line-height: ${rem("22px")};
  font-weight: bold;
  margin: 0 0 6px 0;
`;

const ProjectText = styled.p`
  font-size: ${rem("16px")};
  line-height: ${rem("20px")};
`;

const ImageHolder = styled.div`
  height: 100%;
  width: 100%;
  background: black;
  flex-basis: 50%;
  flex-grow: 1;
`;

const TextWrapper = styled.div`
  flex-basis: 50%;
  padding: 16px 16px;
  flex-grow: 1;
`;

const ImageFit = styled.img`
  height: 100%;
  width: 100%;
  object-fit: cover;
`;

const NoImage = styled.div`
  height: 100%;
  width: 100%;
  background: grey;
`;

const ProjectCard = ({ direction, image, linkedPage, title, description }) => {
  return (
    <ArticalCard direction={direction}>
      <ImageHolder>
        {image ? (
          <ImageFit src={""} alt="Sunset behind mountains." />
        ) : (
          <NoImage />
        )}
      </ImageHolder>
      <TextWrapper>
        <CardHeading1>{title}</CardHeading1>
        <ProjectText>{description}</ProjectText>
        <Link to={linkedPage}>&gt; View Project</Link>
      </TextWrapper>
    </ArticalCard>
  );
};

export default ProjectCard;

ProjectCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  direction: PropTypes.oneOf(["left", "right"]),
  image: PropTypes.string,
  linkedPage: PropTypes.string,
};

ProjectCard.defaultProps = {
  title: "Default Card Title",
  description:
    "This is a description of a project. It contains some detail about the project. This is a description of a project. It contains some detail about the project.",
  image: null,
  linkedPage: "/",
};
