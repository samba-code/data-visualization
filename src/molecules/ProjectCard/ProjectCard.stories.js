import React from "react";

import ProjectCard from "./ProjectCard";

export default {
  title: "Molecules/ProjectCard",
  component: ProjectCard,
  parameters: {
    backgrounds: {
      default: "mid",
      values: [{ name: "mid", value: "#CCC" }],
    },
  },
};

const Template = (args) => <ProjectCard {...args} />;

export const Story1 = Template.bind({});
Story1.args = {};
