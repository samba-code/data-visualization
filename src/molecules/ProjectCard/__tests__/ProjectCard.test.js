import React from "react";
import { render } from "enzyme";
import ProjectCard from "../ProjectCard";

describe("ProjectCard", () => {
  it("ProjectCard to match snapshot", () => {
    const wrapper = render(<ProjectCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
