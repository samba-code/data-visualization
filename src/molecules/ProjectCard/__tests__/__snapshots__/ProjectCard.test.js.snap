// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard ProjectCard to match snapshot 1`] = `
<article
  class="ProjectCard__ArticalCard-qndfbd-0 kCvQUP"
>
  <div
    class="ProjectCard__ImageHolder-qndfbd-3 kDUquc"
  >
    <div
      class="ProjectCard__NoImage-qndfbd-6 jRQeBA"
    />
  </div>
  <div
    class="ProjectCard__TextWrapper-qndfbd-4 cPcrDr"
  >
    <h1
      class="ProjectCard__CardHeading1-qndfbd-1 lfCRVo"
    >
      Default Card Title
    </h1>
    <p
      class="ProjectCard__ProjectText-qndfbd-2 ifIwjv"
    >
      This is a description of a project. It contains some detail about the project. This is a description of a project. It contains some detail about the project.
    </p>
    <a
      aria-current="page"
      href="/"
    >
      &gt; View Project
    </a>
  </div>
</article>
`;
